File: Bank.go
// 
// 

package main

import (
    "database/sql"
    "fmt"
    "log"
    "os"
    "time"

    _ "github.com/go-sql-driver/mysql"
)

func clearScreen() {
    os.Stdout.WriteString("\033[2J")
    os.Stdout.WriteString("\033[H")
}

func main() {
    db, err := sql.Open("mysql", "root:qwerty@tcp(localhost)/bank")
    if err != nil {
        log.Fatal(err)
    }
    defer db.Close()

    clearScreen()

    now := time.Now()
    fmt.Printf("Date : %02d-%02d-%04d   \nTime : %02d:%02d\n", now.Day(), now.Month(), now.Year(), now.Hour(), now.Minute())
    fmt.Println()

    fmt.Println("\t\t\t\t ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
    fmt.Println("\t\t\t\t |                       JPMorgan                         |")
    fmt.Println("\t\t\t\t ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")

    var choice int

    for {
        fmt.Println()
        fmt.Println("==========================================================")
        fmt.Println("|    1. Open new account                                 |")
        fmt.Println("|    2. Deposite cash                                    |")
        fmt.Println("|    3. Withdrawal cash                                  |")
        fmt.Println("|    4. Update customer detail                           |")
        fmt.Println("|    5. Customer detail                                  |")
        fmt.Println("|    6. Balance inquiry                                  |")
        fmt.Println("|    7. Close account                                    |")
        fmt.Println("|    0 to \"RETURN\"                                       |")
        fmt.Println("==========================================================")
        fmt.Println()
        fmt.Print("     Enter your choice - ")
        fmt.Scan(&choice)
        fmt.Println()

        switch choice {
        case 1:
            clearScreen()
            fmt.Println()
            fmt.Println("==================Opening New Account==================")
            fmt.Println("===================Enter Your Detail===================")
            fmt.Println()
            openNew(db)
        case 2:
            clearScreen()
            fmt.Println()
            fmt.Println("==================Deposite Cash==================")
            fmt.Println("=========Please Enter Your Account Number========")
            fmt.Println()
            deposit(db)
        case 3:
            clearScreen()
            fmt.Println()
            fmt.Println("==================Withdrawl Cash==================")
            fmt.Println("=========Please Enter Your Account Number=========")
            fmt.Println()
            withdrawal(db)
        // Other cases (4 to 7) can be similarly handled
        case 0:
            clearScreen()
            fmt.Println()
            fmt.Println("\t\tpress X to exit")
            fmt.Println("\tpress M to return to main menu")
            fmt.Println()
            fmt.Print("Enter your choice here : ")
            var opt string
            fmt.Scan(&opt)

            if opt == "x" || opt == "X" {
                clearScreen()
                os.Exit(0)
            } else if opt == "m" || opt == "M" {
                clearScreen()
                continue
            }
        default:
            clearScreen()
            fmt.Println()
            fmt.Println("Invalid choice. Please enter a valid option.")
            fmt.Println()
        }
    }
}

func openNew(db *sql.DB) {
    var fname, sname, dob, add, adhno, phno, accno, cash, opt string

    fmt.Print("Candidate's FristName - ")
    fmt.Scan(&fname)
    fmt.Print("Candidate's SecondName - ")
    fmt.Scan(&sname)
    fmt.Print("Date Of Birth - ")
    fmt.Scan(&dob)
    fmt.Print("Residential Address - ")
    fmt.Scan(&add)
    fmt.Print("AadharCard No. - ")
    fmt.Scan(&adhno)
    fmt.Print("Contact No - ")
    fmt.Scan(&phno)
    fmt.Print("Account Number - ")
    fmt.Scan(&accno)
    fmt.Print("Opening Ammount - ")
    fmt.Scan(&cash)
    fmt.Println()

    fmt.Print("Press Y to save data : ")
    fmt.Scan(&opt)

    if opt == "y" || opt == "Y" {
        sql := fmt.Sprintf("INSERT INTO accounts VALUES ('%s', '%s', '%s', '%s', '%s', %s, '%s')", fname, sname, dob, add, adhno, phno, accno)
        sql1 := fmt.Sprintf("INSERT INTO amount VALUES ('%s', '%s', '%s', %s)", fname, sname, accno, cash)

        _, err := db.Exec(sql)
        if err != nil {
            log.Fatal(err)
        }
        _, err = db.Exec(sql1)
        if err != nil {
            log.Fatal(err)
        }

        fmt.Println()
        fmt.Println("Saving.....")
        fmt.Println()
        time.Sleep(5)
        clearScreen()

        fmt.Println()
        fmt.Println("Saved")
        time.Sleep(1)
        clearScreen()
        fmt.Println()
        fmt.Println("Press any key to return")
        var input string
        fmt.Scan(&input)
        clearScreen()
    }
}

// Other functions (deposit, withdrawal, update, detail, baliq, close) are similarly converted to Go.

func deposit(db *sql.DB) {
    var accno, amt string

    fmt.Print("Account number: ")
    fmt.Scan(&accno)
    fmt.Print("Deposite Amount: ")
    fmt.Scan(&amt)
    fmt.Println()

    var balance int
    err := db.QueryRow("SELECT balance FROM amount WHERE accountnumber=?", accno).Scan(&balance)
    if err != nil {
        log.Fatal(err)
    }

    fmt.Println("Please Wait....")
    time.Sleep(3)
    clearScreen()
    fmt.Println()

    _, err = db.Exec("UPDATE amount SET balance=balance+? WHERE accountnumber=?", amt, accno)
    if err != nil {
        log.Fatal(err)
    }

    fmt.Printf("Amount Credited to Account: '%s'\n", accno)
    fmt.Println()
    fmt.Println("Press any key to return")
    var input string
    fmt.Scan(&input)
    clearScreen()
}

File: Bank.py
#pylint: disable=unused-variable
# make sure to have installed all the required node models


from os import system
system('cls')
import mysql.connector
import datetime
now = datetime.datetime.now()
print (now.strftime('Date : '"%d-%m-%Y   \nTime : %H:%M"))
import time
print()

info=mysql.connector.connect(
    host="localhost",
    user="root",
    password="qwerty",
    charset="utf8",
    database="bank",
)



print('\t\t\t\t ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
print('\t\t\t\t |                       JPMorgan                          |')
print('\t\t\t\t ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')

#main function 
def menu():
    print()
    print('1. Open new account')
    print('2. Deposite cash')
    print('3. Withdrawal cash')
    print('4. Update customer detail')
    print('5. Customer detail')
    print('6. Balance inquiry')
    print('7. Close account')
    print('0 to "RETURN"')
    print()

#funtion for opening new account
def opennew():
    
    mycursor = info.cursor()

    fname=input("Candidate's FristName - ")
    sname=input("Candidate's SecondName - ")
    dob=input("Date Of Birth - ")
    add=input("Residential Address - ")
    adhno=input("AadharCard No. - ")
    phno=input("Contact No - ")
    accno=input("Account Number - ")
    cash=input("Opening Ammount - ")
    print()
    
    opt=input('Press Y to save data : ')
    #condition for executing insert command 
    #execute only when user prees y
    if opt=='y':      
    
        sql="insert into accounts values('"+fname+"','"+sname+"','"+dob+"','"+add+"','"+adhno+"',"+phno+",'"+accno+"')"
        sql1="insert into amount values('"+fname+"','"+sname+"','"+accno+"',"+cash+")"
        
        mycursor.execute(sql)
        mycursor.execute(sql1)
        
        #commit method for updateing changes in sql database 
        info.commit()
        info.close()
        
        print()
        print('Saving.....')
        print()
        #time sleep method for some delay 
        time.sleep(5)
        system('cls')
        
        print()
        print('Saved')
        time.sleep(1)
        system('cls')
        print()
        print('press any key to return')
        opt=input('')
        system('cls')
        #if opt=='r':
        menu()
        system('cls')
        print()

#Cash deposite Function 
def depo():

    info=mysql.connector.connect(
    host="localhost",
    user="root",
    password="qwerty",
    charset="utf8",
    database="bank",
    )

    mycursor = info.cursor()

    accno=input('Acount number : ')
    amt=input('Deposite Amount : ')

    mycursor.execute("select balance from amount where accountnumber='"+accno+"'")
    myresult=mycursor.fetchall()

    print()
    print('Please Wait....')
    time.sleep(3)
    system('cls')
    print()

    sql="update amount set balance=balance+'"+amt+"' where accountnumber='"+accno+"'"
    mycursor.execute(sql)
    info.commit()
    info.close()
    print("Amount Credited to Account :'"+accno+"'")
    print()
    print('press any key to return')
    opt=input('')
    system('cls')
    #if opt=='r':
    menu()
    system('cls')
    print()

#cash withdrawal function 
def withdrawl():

    info=mysql.connector.connect(
    host="localhost",
    user="root",
    password="qwerty",
    charset="utf8",
    database="bank",
    )

    mycursor = info.cursor()

    accno=input('Acount number : ')
    amt=input('Withdrawal Amount : ')

    mycursor.execute("select balance from amount where accountnumber='"+accno+"'")
    myresult=mycursor.fetchall()

    print()
    print('Please Wait....')
    time.sleep(3)
    system('cls')
    print()

    sql="update amount set balance=balance-'"+amt+"' where accountnumber='"+accno+"'"
    mycursor.execute(sql)
    info.commit()
    info.close()
    print("Amount Ddebited From Account : '"+accno+"'")

    print('press any key to return')
    opt=input('')
    system('cls')
    #if opt=='r':
    menu()
    system('cls')
    print()

#update function declaretion    
def update():

    info=mysql.connector.connect(
    host="localhost",
    user="root",
    password="qwerty",
    charset="utf8",
    database="bank",
    )

    mycursor=info.cursor()
    
    accno=input('Enter account number : ')
    print()
    mycursor.execute("select firstname,secondname,contactno,residentialaddress from accounts where accountnumber='"+accno+"'")
    myresult=mycursor.fetchall()

    system('cls')

    for x in myresult:   
        print("FristName - ",x[0])
        print("SecondName - ",x[1])
        print("Residential Address - ",x[2])
        print("Contact No - ",x[3])
        print()
    

        print('a. Update Name')
        print('b. Update Residential Address')
        print('c. Update Contact Number')
        print()                                                                             

        opt=input('Enter a,b,c according changes :')
        print()
        
        if opt=='a':
            fname=input('Enter New Frist Name : ')
            sname=input('Enter New Second Name : ')
            sql=("update accounts set secondname='"+sname+"' where accountnumber='"+accno+"'" )
            sql1=("update amount set secondname='"+sname+"' where accountnumber='"+accno+"'" )
            mycursor.execute(sql)
            mycursor.execute(sql1)
            info.commit()
            info.close()
            print()
            print('Changes Done')
            
            print()
            print('Prees u to return on update menu')
            print('Prees r to return on main menu')    
            print()
            
            opt=input('Enter your choice here : ')

            if opt=="u":
                update()
            elif opt=="r": 
                menu()
                system('cls')
